server/manager/src/main/java/org/apache/accumulo/manager/upgrade/Upgrader12to13.java

```diff
@@ -347,4 +350,27 @@ private void handlePartialSplits(ServerContext context, String table) {
       throw new IllegalStateException(e);
     }
   }
+
+  private void validateEmptyZKWorkerServerPaths(ServerContext context) {
+    // #4861 added the resource group to the compactor, sserver, tserver
+    // and dead tserver zookeeper paths. Make sure that the these paths
+    // are empty. This means that for the Accumulo 4.0 upgrade, the Manager
+    // should be started first before any other process.
+    final String zkRoot = ZooUtil.getRoot(context.getInstanceID());
+    final ZooReader zr = context.getZooReader();
+    for (String serverPath : new String[] {Constants.ZCOMPACTORS, Constants.ZSSERVERS,
+        Constants.ZTSERVERS, Constants.ZDEADTSERVERS}) {
+      try {
+        List<String> children = zr.getChildren(zkRoot + serverPath);
+        for (String child : children) {
+          if (child.contains(":")) {
```

Comment:
Do we have any limits on what can be in a RG name?  We probably should, not sure if there is.  Wondering if a RG name could contain `:`

Reply:
I don't think we have any limits on the RG name content.